conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -1,7 +1,12 @@
+import json
+
+from conan.tools import CONAN_TOOLCHAIN_ARGS_FILE
 from conan.tools._compilers import architecture_flag, build_type_flags
 from conan.tools.env import Environment
 # FIXME: need to refactor this import and bring to conan.tools
 from conans.client.build.cppstd_flags import cppstd_flag_new
+from conans.client.tools.oss import cross_building, get_cross_building_settings, get_gnu_triplet
```

Comment:
In general try to avoid things from old ``conans.client....`` scope. The ``cross_building, get_cross_building_settings, get_gnu_triplet`` should be copied here, because they are going to be modernized. This one already has a FIXME above, so it can be done in a later PR, but at least the idea for other PRs.

Reply:
okay, let's copy them

Reply:
made a copies of `cross_building`, `get_cross_building_settings`, `get_gnu_triplet` into `conan.tools.oss`. it's like a spider web, one method pulling another one from the `conans.client.tools`, so I had to cut `OSInfo` to only used methods to avoid copying almost everything now (e.g. it requires `which`, `environment_append` and so on for full functionality, it actually does to much - classic god object).

Reply:
yes, this is one of the purposes of doing this copy. All this should be distilled and reduced much, specially if using both profiles, and not needing to detect anything at all. This will follow in future PRs.